src/tutorials/0006/03.md

```diff
@@ -0,0 +1,15 @@
+`CIDv0` uses multihash to support multiple hashing functions. This means that we can successfully generate a hash for specific content using different hashing algorithms and later on be able to identify content using this hash.
+
+But when we're trying to read the data itself, how do we know the encoding of the data?
+It could be encoded with CBOR, Protobuf, plain JSON, etc. Usually, applications solve this issue by trying to guess the encoding based on its content, but this is not a deterministic or safe way to do it.
+
+To solve this issue, `CIDv1` introduces another prefix that uniquely identifies the encoding used.
+
+## Multicodec prefix
+
+The **multicodec prefix** indicates which encoding was used on the data.
+
+![Multicodec Prefix](tutorial-assets/T0006L03-multicodec.jpg)
+
+In the example above we see how data encoded with the codec `dag-pb` would be represented in our CID.
```

Comment:
Ok so, multicodec is just a numeric identifier. They are used for many different purposes. In multihash, the "type" is a multicodec. The multicodec we're talking about here is an "IPLD format". So it's not really correct to title this "Multicodec prefix", although that's techinically what it is, it isn't specific enough. It's like going to see Muse at a gig and the banner outside the venue says "Live tonight: Rock Band" not "Live tonight: Muse".

Reply:
I was under the impression that the multicodec content type on the CID could be _any_ multicodec id. That is why I didn't mention IPLD formats much. So you are saying it can only be an IPLD format id?

Reply:
I guess techincally you could put _any_ multicodec there, but if you want to use the CID with IPFS then it _has_ to be a multicodec for an IPLD format.

Reply:
I understand that in the context of IPFS, IPLDs are used, but since multiformats is project on its own, should we keep this generic just like the CID specification? Or should we contextualize this for IPFS only?

Reply:
The purpose of IPLD is to have interoperable datasets. If you don't use an IPLD format codec then you're doing your own thing and that's not really the future we want to promote.

@terichadbourne should have the final say on this.

Reply:
Thanks for the clarification here, @alanshaw. Sounds like in the context of this tutorial focused on IPFS it would be useful to state that there's a generic thing called a multicodec and that in the case of IPFS we must always use IPLD format as our multicodec. Then for future references list `ipld-format` in the spots Alan has noted below. 

However, worth flagging here that the reason @zebateira referred to it as a multicodec is that he found the multicodec documentation in the multiformats project where CID now resides, and that documentation doesn't mention IPLD format at all. Is that an intentional documentation decision @mikeal (if I'm right that you're off-the-books-kinda-sorta-multiformats-guy in addition to IPLD guy? Is the IPLD format multicodec described somewhere else instead? Could we link between the two places in the docs?

FWIW I find it sooooooo confusing to be a learner in this ecosphere popping around in different repos of related projects. I think @zebateira  actually found the issue in which CID got moved from IPLD to Multiformats in his quest to find the right docs, so he deserves a detective magnifying glass trophy.

Reply:
added at the end of the lesson: `To note that, within IPFS, only IPLD formats are used so that means the multicodec prefix will always be an IPLD codec.`

Reply:
Let me try to simplify this a little bit.

`multicodec` is just a table of identifiers. There isn’t an enforced classification between the identifiers (informally we have some comment and ranges in the table), some are hash algorithms, some are serializable formats, and when you get into encryption they are potentially both.

A CID is just a multicodec prefix for the **codec** plus a multihash (which has another multicodec prefix for the hash algorithm), there isn’t much validation here for which part of the table the prefix is pulled from. You could create a CID with a multicodec prefix for a hash algorithm and it won’t end up failing until you try to lookup the codec implementation.

I wouldn’t use the term “IPLD format” because that’s also rather confusing. IPLD is about building data structures out links to any data you can address with a CID and that includes codecs that really have nothing to do with IPLD (JSON, git, etc). Even the term “IPLD codec” implies that these are built by/for IPLD when in many cases they are not.

I would just use the term **“codec.”** The main difference between the identifier used in the CID and the one used for multihash is that the data being addressed by the CID should be parsable by a codec for the given the identifier. That’s it. There’s nothing IPLD specific about CID’s and their codec identifiers.

Being that IPLD isn’t a single project but an umbrella for things we do with content addressed data structures, it’s often hard to figure out where the lines are and when something should be called “IPLD.” CID is right on the edge because it’s what we use to link data together and once you’re liking data together, that’s IPLD. However, you *could* use CID’s **without linking anything together**. You could have a database of hashed JSON blocks and all the id’s are CID’s with a JSON multicodec prefix. Perfectly valid use of CID, but nothing is ever linked together, so it’s not really “IPLD.”